### PR TITLE
Add release changelog

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds the `v1.0.1` release changelog.

## Related Issues

#382 

## Types of Changes

- [x] Documentation

### CODE REVIEW

File: docs/release_notes/v1.0.1.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+[id="{p}-release-notes-v1.0.1"]
+=== Elastic Cloud Control (ecctl) v1.0.1
+++++
+<titleabbrev>v1.0.1</titleabbrev>
+++++
+
+<<{p}-release-notes-v1.0.1-bug-fixes,Bug fixes>> | <<{p}-release-notes-v1.0.1-changelog,Changelog>>
+
+Welcome to the v1.0.1 release of {n}. This version brings a bug fix.
```
Comment (alaudazzi) [inline docs/release_notes/v1.0.1.adoc:9]:
```suggestion
{n} version 1.0.1 includes a bug fix.
```

File: docs/release_notes/v1.0.1.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Elastic Cloud Control (ecctl) is Elastic’s CLI to manage Elasticsearch Service (ESS) and Elastic Cloud Enterprise (ECE).
```
Comment (alaudazzi) [inline docs/release_notes/v1.0.1.adoc:11]:
```suggestion
Elastic Cloud Control (ecctl) is the CLI for Elasticsearch Service (ESS) and Elastic Cloud Enterprise (ECE).
```

**Review by alaudazzi — APPROVED** [review]
Minor edits. otherwise LGTM.

**Review by karencfv — COMMENTED** [review]
Looks good! Can you add the changes @alaudazzi requested to the templates as well?

**Pull request merged**